### ISSUE TITLE
`clippy::from_over_into` produces incorrect suggested code for unit type

### ISSUE THREAD

**Issue by ctz** [rust-lang/rust-clippy#12138]
### Summary

```
struct Hello;

impl Into<()> for Hello {
    fn into(self) {
    }
}
```

Triggers `clippy::from_over_into` but the suggested fix is syntactically invalid.

### Reproducer

I tried this code:

```rust
struct Hello;

impl Into<()> for Hello {
    fn into(self) {
    }
}

fn main() {
    Hello.into()
}
```

I expected to see this happen -- perhaps:

```
warning: an implementation of `From` is preferred since it gives you `Into<_>` for free where the reverse isn't true
 --> src/main.rs:3:1
  |
3 | impl Into<()> for Hello {
  | ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: `impl From<Local> for Foreign` is allowed by the orphan rules, for more information see
          https://doc.rust-lang.org/reference/items/implementations.html#trait-implementation-coherence
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#from_over_into
  = note: `#[warn(clippy::from_over_into)]` on by default
help: replace the `Into` implementation with `From<Hello>`
  |
3 ~ impl From<Hello> for () {
4 ~     fn from(_: Hello) {
  |
```

Instead, this happened:

```
warning: an implementation of `From` is preferred since it gives you `Into<_>` for free where the reverse isn't true
 --> src/main.rs:3:1
  |
3 | impl Into<()> for Hello {
  | ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: `impl From<Local> for Foreign` is allowed by the orphan rules, for more information see
          https://doc.rust-lang.org/reference/items/implementations.html#trait-implementation-coherence
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#from_over_into
  = note: `#[warn(clippy::from_over_into)]` on by default
help: replace the `Into` implementation with `From<Hello>`
  |
3 ~ impl From<Hello> for () {
4 ~     fn from(val: Hello)Self {
  |
```

Note the last line: `fn from(val: Hello)Self {` is not valid rust syntax.

### Version

```text
rustc 1.75.0 (82e1608df 2023-12-21)
binary: rustc
commit-hash: 82e1608dfa6e0b5569232559e3d385fea5a93112
commit-date: 2023-12-21
host: x86_64-unknown-linux-gnu
release: 1.75.0
LLVM version: 17.0.6
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
